site/src/comparison.rs

```diff
@@ -411,47 +411,70 @@ pub fn write_summary_table(
             .unwrap_or_else(|| "N/A".to_string())
     }
 
-    writeln!(
-        result,
-        r#"| | Regressions 😿 <br />(primary) | Regressions 😿 <br />(secondary) | Improvements 🎉 <br />(primary) | Improvements 🎉 <br />(secondary) | All 😿 🎉 <br />(primary) |
-|:---:|:---:|:---:|:---:|:---:|:---:|"#
-    )
-    .unwrap();
-    writeln!(
-        result,
-        "| count{} | {} | {} | {} | {} | {} |",
-        if with_footnotes { "[^1]" } else { "" },
-        primary.num_regressions,
-        secondary.num_regressions,
-        primary.num_improvements,
-        secondary.num_improvements,
-        primary.num_regressions + primary.num_improvements
-    )
-    .unwrap();
-
-    writeln!(
-        result,
-        "| mean{} | {} | {} | {} | {} | {} |",
-        if with_footnotes { "[^2]" } else { "" },
-        render_stat(primary.num_regressions, || Some(
-            primary.arithmetic_mean_of_regressions()
-        )),
-        render_stat(secondary.num_regressions, || Some(
-            secondary.arithmetic_mean_of_regressions()
-        )),
-        render_stat(primary.num_improvements, || Some(
-            primary.arithmetic_mean_of_improvements()
-        )),
-        render_stat(secondary.num_improvements, || Some(
-            secondary.arithmetic_mean_of_improvements()
-        )),
+    let columns = [
+        "          ", // we want at least 10 spaces to accommodate "count[^1]"
+        "Regressions 😿 <br />(primary)",
+        "Regressions 😿 <br />(secondary)",
+        "Improvements 🎉 <br />(primary)",
+        "Improvements 🎉 <br />(secondary)",
+        "All 😿 🎉 <br />(primary)",
+    ];
+    let counts: Vec<usize> = columns
+        .iter()
+        .map(|s| {
+            // Unicode emojis usually have width equal to ~2 chars (maybe? :) ).
+            let count = s.chars().count();
+            if s.chars().any(|c| !c.is_whitespace()) {
+                count + 1
+            } else {
+                count
+            }
+        })
+        .collect();
```

Comment:
Do we need to collect this since it will just be used in the call to `zip`?

Reply:
It's used in two places, so it has to be collected into a `Vec` (afaik you can't really reuse an iterator in a generic way?).